runtime/lua/vim/lsp.lua

```diff
@@ -50,6 +71,16 @@ local function resolve_bufnr(bufnr)
   return bufnr
 end
 
+--@private
+--- callback called by the client when trying to call a method that's not
+--- supported in any of the servers registered for the current buffer.
+--@param method (string) name of the method
+function lsp._unsupported_method(method)
+  local msg = string.format("method %s is not supported by any of the servers registered for the current buffer", method)
+  local _ = log.warn() and log.warn(msg)
```

Comment:
you can just do `log.warn(msg)` -- you only need to do the check if it is expensive to calculate `msg`.